### PR TITLE
Timetable Foundation

### PR DESCRIPTION
**Pull request by jeppekroghitk**
#### Link to ticket

N/A

#### Description

Foundation for further development

#### Screenshot of the result

N/A

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: Controllers/TimeTable.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * get
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     */
+    public function get(): Response
+    {
+        $timeTableStyle = dirname(dirname($_SERVER["DOCUMENT_ROOT"])) . "dist/css/timeTable.css";
```
Comment (sinejespersen) [inline Controllers/TimeTable.php]: ```dirname(dirname``` creeps me a bit out. can it be separated into variables in a way?

File: Services/TimeTable.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if (file_exists($target)) {
+                unlink($target);
+            }
+            symlink($source, $target);
+        }
+    }
+
+    public function uninstall(): void
+    {
+        // Remove tables
```
Comment (sinejespersen) [inline Services/TimeTable.php]: remove assets?

File: Templates/settings.blade.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+@extends($layout)
+
+@section('content')
+
+<x-global::pageheader :icon="'fa fa-puzzle-piece'">
+    <h1>OmniSearch Settings</h1>
```
Comment (sinejespersen) [inline Templates/settings.blade.php]:
```suggestion
    <h1>Time table Settings</h1>
```

File: Templates/settings.blade.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+</x-global::pageheader>
+
+
+<div class="maincontent">
+<?php echo $tpl->displayNotification(); ?>
+    <div class="maincontentinner">
+        <h5 class="subtitle">Omnisearch Settings</h5>
```
Comment (sinejespersen) [inline Templates/settings.blade.php]:
```suggestion
        <h2 class="subtitle">Time table Settings</h2>
```

File: Templates/settings.blade.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        <p style="padding-bottom: 15px">These settings will change the way the OmniSearch plugin works.</p>
```
Comment (sinejespersen) [inline Templates/settings.blade.php]:
```suggestion
        <p style="padding-bottom: 15px">These settings will change the way the Time table plugin works.</p>
```

File: Templates/settings.blade.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        <h4 class="widgettitle title-light"><span class="fa fa-cog"></span>
+            Cache expiration durations
+        </h4>
+        <form class="" method="post" id="" action="<?=BASE_URL ?>/OmniSearch/settings">
```
Comment (sinejespersen) [inline Templates/settings.blade.php]: Maybe strip this file from omnisearchstuff

File: register.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+use Leantime\Core\Events;
+
+/**
+ * Adds a menu point for adding fixture data.
+ * @param array<string, array<int, array<string, mixed>>> $menuStructure The existing menu structure to which the new item will be added.
+ * @return array<string, array<int, array<string, mixed>>> The modified menu structure with the new item added.
+ */
+function addImportDataMenuPointTimeTable(array $menuStructure): array
+{
+    $menuStructure['personal'][16] = [
```
Comment (sinejespersen) [inline register.php:14]: add a comment about this myserious 16

**Review by sinejespersen — CHANGES_REQUESTED** [review]
Some small changes....

File: Templates/settings.blade.php (hunk 6 above)
Comment (jeppekroghitk) [reply Templates/settings.blade.php]: Yeah. Thought to keep it as a reference for how to set settings up nicely, but makes less sense to keep now in git.

File: register.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    return $menuStructure;
+}
+
+/**
+ * Adds project overview to the personal menu
+ * @return string - the string "personal" if the route is ProjectOverview.projectOverview.
+ */
+function addProjectOverviewToPersonalMenuTimeTable(): string
+{
+    if (FrontcontrollerCore::getCurrentRoute() === 'ProjectOverview.projectOverview') {
```
Comment (sinejespersen) [inline register.php]:
```suggestion
    if (FrontcontrollerCore::getCurrentRoute() === 'TimeTable.timetable) {
```

**Pull request merged**